src/io_uring.c

```diff
@@ -0,0 +1,92 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
+
+void initIOUring(void) {
+    if (server.io_uring_enabled) {
+        struct io_uring_params params;
+        struct io_uring *ring = zmalloc(sizeof(struct io_uring));
+        memset(&params, 0, sizeof(params));
+        /* On success, io_uring_queue_init_params(3) returns 0 and ring will
+         * point to the shared memory containing the io_uring queues.
+         * On failure -errno is returned. */
+        int ret = io_uring_queue_init_params(IO_URING_DEPTH, ring, &params);
+        if (ret != 0) {
+            /* Warning if user enable the io_uring in config but system doesn't support yet. */
+            serverLog(LL_WARNING, "System doesn't support io_uring, disable io_uring.");
+            zfree(ring);
+            server.io_uring = NULL;
+        } else {
+            serverLog(LL_NOTICE, "System support io_uring, enable io_uring.");
```

Comment:
No need to mention that system supports it? If it can be enabled, it means it's supported.

```suggestion
            serverLog(LL_NOTICE, "io_uring enabled.");
```